src/chi/javascript/components/accordion.js

```diff
@@ -0,0 +1,161 @@
+    });
+  }
+
+  _resizeParentAccordions(e, action) {
+    const resizedAccordionContent = e.querySelector(`.${CONTENT_CLASS}`);
+    const resizedAccordionContentHeight = parseInt(Util.calculateHiddenElementHeight(resizedAccordionContent));
+
+    for (let cur = e.parentNode;
+      cur && cur.nodeName !== 'BODY';
+      cur = cur.parentNode) {
+        if (cur.classList.contains(COMPONENT_CLASS)) {
+          const content = cur.querySelector(`.${CONTENT_CLASS}`);
+          const contentHeight = parseInt(content.style.height);
+          let newHeight;
+
+          if (action === 'show') {
+            newHeight = contentHeight + resizedAccordionContentHeight;
+          } else if (action === 'hide') {
+            newHeight = contentHeight - resizedAccordionContentHeight;
+          }
+
+          content.style.height = `${newHeight}px`;
+      }
+      if (cur.classList.contains(COMPONENT_CLASS) &&
+        !cur.classList.contains(CHILD_ACCORDION_CLASS)) {
+          break;
+      }
+    }
+  }
+
+  _initInnerAccordions() {
+    const self = this;
+    const childAccordions = self._content.getElementsByClassName(COMPONENT_CLASS);
+
+    if (childAccordions) {
+      Array.prototype.forEach.call(
+        childAccordions,
+        elem => {
+          const trigger = elem.querySelector(`.${TRIGGER_CLASS}`);
+          const content = elem.querySelector(`.${CONTENT_CLASS}`);
+  
+          if (trigger && content) {
+            const newAccordion = Accordion.factory(elem, DEFAULT_CONFIG);
+  
+            this._childAccordions.push(newAccordion);
+            Util.addClass(elem, CHILD_ACCORDION_CLASS);
+          }
+        }
+      );
+    }
+  }
+
+  static get componentType() {
+    return COMPONENT_TYPE;
+  }
+
+  show() {
+    const contentElem = this._content;
+    let contentHeight = Util.calculateHiddenElementHeight(contentElem);
+
+    if (!this._expanded) {
+      this._resizeParentAccordions(this._elem, 'show');
+      this._elem.dispatchEvent(
+        Util.createEvent(EVENTS.SHOW)
+      );
+      
+      Util.threeStepsAnimation(
+        () => {
+          Util.addClass(contentElem, chi.classes.TRANSITIONING);
+          contentElem.style.height = '0px';
+          contentElem.style.opacity = '0.5';
+        }, () => {
+          Util.addClass(this._elem, chi.classes.EXPANDED);
+          contentElem.style.height = contentHeight;
+          contentElem.style.opacity = '1';
+        }, () => {
+          Util.removeClass(contentElem, chi.classes.TRANSITIONING);
+        }, 75
+      );
+
+      this._expanded = true;
+    }
+  }
+
+  hide() {
+    const contentElem = this._content;
+
+    if (this._expanded) {
+      this._resizeParentAccordions(this._elem, 'hide');
+      this._elem.dispatchEvent(
+        Util.createEvent(EVENTS.HIDE)
+      );
+      Util.threeStepsAnimation(
+        () => {
+          Util.checkAddClass(contentElem, chi.classes.TRANSITIONING);
+          contentElem.style.opacity = '1';
```

Comment:
Please, set back the initial height of the animation (if you remove it when the animation finishes as I asked you in my previous comment).